dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/validation/AppValidator.java

```diff
@@ -0,0 +1,180 @@
+     */
+    public AppValidator withClass(Class<?> clazz) {
+        classes.add(clazz);
+        return this;
+    }
+
+    /**
+     * Specify that the app should fail to start, and that the given failure message should be seen during app startup
+     */
+    public AppValidator failsWith(String expectedFailure) {
+        stringsToFind.add(expectedFailure);
+        stringsToFind.add(APP_FAIL_CODE);
+        server.addIgnoredErrors(Collections.singletonList(expectedFailure));
+        return this;
+    }
+
+    /**
+     * Specify that the app should start, but that the given message should be seen during app startup
+     * <p>
+     * This can be used to ensure that warning messages are emitted
+     */
+    public AppValidator succeedsWith(String expectedMessage) {
+        stringsToFind.add(expectedMessage);
+        stringsToFind.add(APP_START_CODE);
+        server.addIgnoredErrors(Collections.singletonList(expectedMessage));
+        return this;
+    }
+
+    public AppValidator withAppConfig(PropertiesAsset appConfig) throws Exception {
+        this.appConfig = appConfig;
+        return this;
+    }
+
+    public AppValidator withAppName(String appName) {
+        this.appName = appName;
+        return this;
+    }
+
+    /**
+     * Specify that the app should start successfully
+     * <p>
+     * This can be used to ensure that valid edge cases do not cause a validation failure
+     */
+    public AppValidator succeeds() {
+        stringsToFind.add(APP_START_CODE);
+        return this;
+    }
+
+    /**
+     * Run the validation
+     * <p>
+     * This consists of the following steps
+     * <ul>
+     * <li>Build the app archive from the classes given using {@link #withClass(Class)}</li>
+     * <li>Deploy the app to the server passed to {@link #validateAppOn(LibertyServer)}</li>
+     * <li>if {@link #failsWith(String)} was called, validate that any expected messages are found in the log and that the app fails to start</li>
+     * <li>if {@link #succeedsWith(String)} was called, validate that any expected messages are found in the log and that the app starts successfully</li>
+     * <li>if {@link #succeeds()} was called, validate that the app starts successfully</li>
+     * <li>Remove the app from the server</li>
+     * </ul>
+     */
+    public void run() {
+        try {
+            String archiveName;
+
+            if (appName != null) {
+                archiveName = appName + ".war";
+            } else {
+                archiveName = "testApp-" + (++appCount) + ".war";
+            }
+
+            try {
+                WebArchive app = ShrinkWrap.create(WebArchive.class, archiveName);
+
+                for (Class<?> clazz : classes) {
+                    app = app.addClass(clazz);
+                }
+                if (appConfig != null) {
+                    app = app.addAsLibraries(kafkaClientLibs())
```

Comment:
I would have a generic methods on the class that could add libraries and manifests to the app, so that we don't have to include kafka specific entities within an application that might not use kakfka.

Also putting this in the appConfig block when they are unrelated doesn't make sense.